pkg/kvstore/kvstore.go

```diff
@@ -32,54 +34,59 @@ var (
 // This package implements a very basic key/value store.  It is up to the
 // caller to provision the namespace.
 
-type KeyValueStore struct {
-	b Backend
-
-	kv map[string][]byte
-
-	fileName string
-
-	l sync.RWMutex
+type KeyValueStore interface {
+	Set(ctx context.Context, key string, value []byte) error
+	Get(key string) ([]byte, error)
+	List(re string) (map[string][]byte, error)
+	Delete(ctx context.Context, key string) error
+	Save(ctx context.Context) error
+	Name() string
```

Comment:
What is `Name` for?  Comment would be good since the rest, I think, are self explanatory.